video/out/wldmabuf/ra_wldmabuf.c

```diff
@@ -30,33 +28,18 @@ static void destroy(struct ra *ra)
     talloc_free(ra->priv);
 }
 
-bool ra_compatible_format(struct ra *ra, int imgfmt, uint32_t drm_format, uint64_t modifier)
+bool ra_compatible_format(struct ra *ra, uint32_t drm_format, uint64_t modifier)
 {
     struct priv *p = ra->priv;
     struct vo_wayland_state *wl = p->vo->wl;
-    struct drm_format *formats = wl->compositor_formats;
-
-    // Always check if the compositor supports the format.
-    bool supported_compositor_format = false;
-    for (int i = 0; i < wl->num_compositor_formats; ++i) {
-        if (formats[i].format != drm_format || formats[i].modifier == DRM_FORMAT_MOD_INVALID)
-            continue;
-        if (modifier == formats[i].modifier)
-            return true;
-        supported_compositor_format = true;
-    }
 
-    if (!supported_compositor_format)
-        return false;
+    // First check if any scanout tranches exist with a format match.
+    if (vo_wayland_format_in_scanout(wl, drm_format, modifier))
+        return true;
```

Comment:
So in that case is it safe to only care about the main device? I was thinking maybe it might be possible for a compositor to advertise a scanout tranche to us on a different device and we should prefer that if the format is there. AFAIK, compositors do actually scan out this VO eg like in issue #14592. But whether or not they do so is of course out of our control.

Reply:
>So in that case is it safe to only care about the main device?

I have not looked at how this PR deals with multiple devices. But the protocol says that tranches are sent in the order of preference.

Reply:
Maybe I'm overthinking this. In the multi GPU situation, perhaps mpv just needs to pick the first usable tranche we get? Whatever device it happens to be doesn't matter for us? If so, I need to invert the structs so tranches is the outer loop and not loop over devices.

Reply:
Probably.

Reply:
Thanks, that makes sense. I can simplify this quite a bit then.